### ISSUE TITLE
LuisDialog: Response status code does not indicate success: 403 (Quota Exceeded).

### ISSUE THREAD

**Issue by KIBASSA** [Microsoft/BotBuilder#2298]
Hi, 
Today without making any manipulation on my class LuisModel or in my application in www.luis.ai, at about 1 pm (Paris time), I obtained this error:

System.Net.Http.HttpRequestException:
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Luis.LuisService+<Microsoft-Bot-Builder-Luis-ILuisService-QueryAsync>d__4.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Luis.Extensions+<QueryAsync>d__3.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.LuisDialog`1+<MessageReceived>d__7.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.DialogTask+ThunkResume`1+<Rest>d__4.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Internals.Fibers.Wait`2+<Microsoft-Bot-Builder-Internals-Fibers-IWait<C>-PollAsync>d__19.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Internals.Fibers.Frame`1+<Microsoft-Bot-Builder-Internals-Fibers-IFrameLoop<C>-PollAsync>d__7.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Internals.Fibers.Fiber`1+<Microsoft-Bot-Builder-Internals-Fibers-IFiberLoop<C>-PollAsync>d__13.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Internals.Fibers.Wait`2.Microsoft.Bot.Builder.Internals.Fibers.IAwaiter<T>.GetResult (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at Microsoft.Bot.Builder.Dialogs.Chain+LoopDialog`1+<ResumeAsync>d__3.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.DialogTask+ThunkResume`1+<Rest>d__4.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Internals.Fibers.Wait`2+<Microsoft-Bot-Builder-Internals-Fibers-IWait<C>-PollAsync>d__19.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Internals.Fibers.Frame`1+<Microsoft-Bot-Builder-Internals-Fibers-IFrameLoop<C>-PollAsync>d__7.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Internals.Fibers.Fiber`1+<Microsoft-Bot-Builder-Internals-Fibers-IFiberLoop<C>-PollAsync>d__13.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.DialogTask+<Microsoft-Bot-Builder-Dialogs-Internals-IDialogStack-PollAsync>d__19.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.DialogTask+<Microsoft-Bot-Builder-Dialogs-Internals-IPostToBot-PostAsync>d__21`1.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.ReactiveDialogTask+<Microsoft-Bot-Builder-Dialogs-Internals-IPostToBot-PostAsync>d__5`1.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.ExceptionTranslationDialogTask+<Microsoft-Bot-Builder-Dialogs-Internals-IPostToBot-PostAsync>d__2`1.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.LocalizedDialogTask+<Microsoft-Bot-Builder-Dialogs-Internals-IPostToBot-PostAsync>d__2`1.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.ScoringDialogTask`1+<Microsoft-Bot-Builder-Dialogs-Internals-IPostToBot-PostAsync>d__6`1.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.PersistentDialogTask+<Microsoft-Bot-Builder-Dialogs-Internals-IPostToBot-PostAsync>d__6`1.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at Microsoft.Bot.Builder.Dialogs.Internals.PersistentDialogTask+<Microsoft-Bot-Builder-Dialogs-Internals-IPostToBot-PostAsync>d__6`1.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.SerializingDialogTask+<Microsoft-Bot-Builder-Dialogs-Internals-IPostToBot-PostAsync>d__4`1.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Internals.PostUnhandledExceptionToUserTask+<Microsoft-Bot-Builder-Dialogs-Internals-IPostToBot-PostAsync>d__5`1.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at Microsoft.Bot.Builder.Dialogs.Internals.PostUnhandledExceptionToUserTask+<Microsoft-Bot-Builder-Dialogs-Internals-IPostToBot-PostAsync>d__5`1.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Conversation+<SendAsync>d__4.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Microsoft.Bot.Builder.Dialogs.Conversation+<SendAsync>d__2.MoveNext (Microsoft.Bot.Builder, Version=3.2.0.0, Culture=neutral, PublicKeyToken=31bf3856ad364e35)
   at System.Runtime.CompilerServices.TaskAwaiter.ThrowForNonSuccess (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification (mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089)
   at Emiko.MessagesController+<Post>d__0.MoveNext (BotUI, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null)

I have no idea how I can make to solve that.
And I am completely blocked.

Anybody has an idea how to solve this problem?

Thanks,
KIBASSA

**Comment by andradedearthur** [comment]
Are you using the free plan? The free subscription has a limit of [10.000 calls per month](https://www.microsoft.com/cognitive-services/en-us/language-understanding-intelligent-service-luis). You should check if you have reached this limit.

**Comment by KIBASSA** [comment]
Yes, it was a free use.
I have just activated a standard subscription.
I have to copy the key generated in portal azure

![image](https://cloud.githubusercontent.com/assets/8443316/23097074/063c8b14-f62b-11e6-88ca-2bc81e76f0d1.png)



![image](https://cloud.githubusercontent.com/assets/8443316/23097107/c559ee60-f62b-11e6-81ad-bb5391823468.png)

Then I pasted this into my applcation of luis.ai

I always have the error.
There is another manipulation to be made?

**Comment by KIBASSA** [comment]
After sending a message to the support, they asked me to update my published model and use the new endpoint.

Once I did that, everything worked well.

Junior

**Comment by willportnoy** [comment]
I'm glad you were able to resolve the issue - I'm going to close it for now.

**Comment by anirudhaditya** [comment]
Kibassa, I am also facing the same issue. I have updated the API from https://api.projectoxford.ai/luis/v1/application, to 
Current API: https://westus.api.cognitive.microsoft.com/luis/v2.0/apps still am facing issues with Exceed Quota. I have full subscription. What does "Update my published model" means. I do not see any option to update the model.

**Comment by madsamuel** [comment]
I had to drop Ocp-Apim-Subscription-Key from the header when making the call to the endpoint.

**Comment by managedcoder** [comment]
I had this same problem recently and the UI in the portals has changed but the overall process of linking a luis service from the Azure portal to a LUIS app in the luis.ai portal is still:
	1. Create luis service in Azure portal choosing pricing tier that suits your quota and throttling needs
	2. Create new app in luis.ai portal or select existing one and link the luis app to the luis service (see screen grab below).
	3. Grab the new subscription key by clicking Key 1 in newly added linked resource in list and use it to update subscription key in your app
	4. Update, if necessary, location portion of the endpoint Url with the Region value shown in the newly added linked resource
You LUIS App ID remains the same so once you change the above you should be back in business.
![image](https://user-images.githubusercontent.com/17769186/51300155-e967f980-19f8-11e9-8b6f-99281ddebc01.png)

**Comment by bnj123** [comment]
@managedcoder I am facing the same problem. Have you managed to solve it?

**Comment by managedcoder** [comment]
@bnj123, I did solve this by linking a LUIS service from my Azure subscription which I outlined in my previous post.  In my case, I had used the Authoring Key to create the LUIS service in the LUIS portal but that key is free to use but it has a cap/quota that I exceeded so the solution was to create a "paid" endpoint by creating the LUIS app in my Azure subscription and because its a billed service there is no count-based cap but there is a request per second cap which was totally fine for my customer.

So, if you are in the same predicament, follow up instructions in my last post and you will no longer hit the quota error.

-Russ
